src/meta/client/MetaClient.cpp

```diff
@@ -39,8 +39,24 @@ MetaClient::~MetaClient() {
 }
 
 
-void MetaClient::init() {
+void MetaClient::waitForMetadReady() {
     loadDataThreadFunc();
+
+    if (!ready_ && sendHeartBeat_) {
+        while (true) {
+            if (listener_ == nullptr) {
+                VLOG(1) << "Can't send heartbeat due to listener_ is nullptr!";
+            }
+            auto ret = heartbeat().get();
+            if (!ret.ok()) {
+                LOG(ERROR) << "Heartbeat failed, status:" << ret.status();
+                ::sleep(1);
```

Comment:
maybe can use std::this_thread::sleep_for to sleep short

Reply:
fixed! Thanks!